mycroft/skills/phillips_hue/__init__.py

```diff
@@ -0,0 +1,283 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Mycroft Core.  If not, see <http://www.gnu.org/licenses/>.
+
+from adapt.intent import IntentBuilder
+from mycroft.skills.core import MycroftSkill
+from mycroft.util.log import getLogger
+from os.path import dirname
+from phue import Bridge
+from phue import Group
+from phue import PhueRegistrationException
+from phue import PhueRequestTimeout
+import socket
+import time
+
+__author__ = 'ChristopherRogers1991'
+
+LOGGER = getLogger(__name__)
+
+
+class DeviceNotFoundException(Exception):
+    pass
+
+
+class PhillipsHueSkill(MycroftSkill):
+
+    def __init__(self):
+        super(PhillipsHueSkill, self).__init__(name="PhillipsHueSkill")
+        self.brightness_step = self.config.get('brightness_step', '')
+        if self.brightness_step == '':
+            self.brightness_step = 50
+        else:
+            self.brightness_step = int(self.brightness_step)
+        self.color_temperature_step =\
+            self.config.get('color_temperature_step', '')
+        if self.color_temperature_step == '':
+            self.color_temperature_step = 1000
+        else:
+            self.color_temperature_step = int(self.color_temperature_step)
+        self.verbose = True if self.config.get('verbose') == 'True' else False
+        self.username = self.config.get('username', '')
+        if self.username == '':
+            self.username = None
+        self.ip = None  # set in _connect_to_bridge
+        self.bridge = None
+        self.all_lights = None
+
+    @property
+    def connected(self):
+        return self.bridge is not None
+
+    def _connect_to_bridge(self):
+        try:
+            self.ip = self.config.get('ip', '')
+            if self.ip == '':
+                self.ip = _discover_bridge()
+            self.bridge = Bridge(self.ip, self.username)
+            self.username = self.bridge.username
+        except DeviceNotFoundException:
+            self.speak_dialog('bridge.not.found')
+            return False
+        except PhueRegistrationException:
+            self.speak_dialog('connect.to.bridge')
+            i = 0
+            while i < 30:
+                time.sleep(1)
+                try:
+                    self.bridge = Bridge(self.ip)
+                except PhueRegistrationException:
+                    continue
+                else:
+                    break
+            if not self.connected:
+                self.speak_dialog('failed.to.register')
+                return False
+            else:
+                self.speak_dialog('successfully.registered')
+        self.all_lights = Group(self.bridge, 0)
+        return True
+
+    def initialize(self):
+        self.load_data_files(dirname(__file__))
+
```

Comment:
When changing to use regex, you'll also need to have a `load_regex_files` call here. Look at the [Stock skill](https://github.com/MycroftAI/mycroft-core/blob/master/mycroft/skills/stock/__init__.py) for examples.